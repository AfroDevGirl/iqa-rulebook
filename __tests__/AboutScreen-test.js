import React from 'react'
import AboutScreenView from '../screens/AboutScreen/AboutScreen.js'
import renderer from 'react-test-renderer'

describe('AboutScreenView', () => {
  test('renders correctly', () => {
    const screen = renderer.create(<AboutScreenView />).toJSON()
    expect(screen).toMatchSnapshot()
  })
})
