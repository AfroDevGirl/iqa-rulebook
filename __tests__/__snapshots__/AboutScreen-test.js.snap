// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutScreenView renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
    ]
  }
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "column-reverse",
        },
        Object {
          "backgroundColor": "#000",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={undefined}
        pointerEvents="auto"
        style={
          Object {
            "backgroundColor": "#E9E9EF",
            "bottom": 0,
            "left": 0,
            "marginTop": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 5,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <RCTScrollView>
          <View>
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "fontSize": 20,
                      "fontWeight": "bold",
                    }
                  }
                >
                  Mission
                </Text>
              </View>
              <View
                style={
                  Object {
                    "padding": 10,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  The IQA is responsible for the governance and sustainability of the sport of quidditch and provides responsive and effective leadership to support the sport's development throughout the world. Through the sport of quidditch, the IQA endeavors to improve gender education across all sports and communities, promote equality and diversity, and foster a love of reading across all ages. The organization works to promote youth engagement, leadership, and physical activity amongst players and potential players wherever they are found.
                </Text>
              </View>
            </View>
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "fontSize": 20,
                      "fontWeight": "bold",
                    }
                  }
                >
                  History
                </Text>
              </View>
              <View
                style={
                  Object {
                    "padding": 10,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  Real-life quidditch was created on a sunny Sunday afternoon by Xander Manshel and Alex Benepe, students at Middlebury College. In 2005 they began playing regular intramural games with their fellow students. Two years later the game spread beyond Middlebury and the first intercollegiate match was organized. The fun weekend pastime laid the foundations for the full contact, gender inclusive, and highly competitive sport played today.
                </Text>
              </View>
              <View
                style={
                  Object {
                    "padding": 10,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  2008 saw a 13 team World Cup championship take place in Middlebury. A team from McGill University of Quebec became the first non - US competitor to join the sport. As more schools created teams, it became clear that the sport needed more formal governance. The International Quidditch Association was created to oversee the future quidditch growth and incorporated as a nonprofit business in 2010.
                </Text>
              </View>
              <View
                style={
                  Object {
                    "padding": 10,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  In 2011 the IQA organized World Cup V in New York City. This time it fielded a 96 - team tournament including competitors from the United States, Canada, and Finland. Over 10,000 spectators attended the event. The same year, the newly formed Australian Quidditch Association organized its first major tournament. The QUAFL Cup was held in New South Wales and became the first quidditch event to take place outside of North America.
                </Text>
              </View>
              <View
                style={
                  Object {
                    "padding": 10,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  In 2012 the IQA hosted national teams in Oxford, England, for the inaugural Global Games. Competitors included United States, Canada, United Kingdom, France, and Australia.The Games became a bi - annual tournament showcasing the talents of the best players from around the world. A unique exhibition match occurred alongside a torch ceremony for the London Olympics that year. The Games spurred interest and growth in quidditch across the globe.
                </Text>
              </View>
              <View
                style={
                  Object {
                    "padding": 10,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  The second Global Games took place in 2014 in Burnaby, British Columbia. Belgium and Mexico joined the five previous competitors. The same year the IQA was relaunched as US Quidditch to manage the sport in the US. A dedicated transition team was established to build a new IQA with a solely international focus.
                </Text>
              </View>
              <View
                style={
                  Object {
                    "padding": 10,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  In 2016 the third Global Games were renamed to the IQA World Cup. The competition took place in Frankfurt, Germany, and featured 21 teams from four different continents. Australia won their first championship, beating the previous double champions, the United States.
                </Text>
              </View>
              <View
                style={
                  Object {
                    "padding": 10,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                >
                  The IQA serves approximately 20 quidditch-playing organizations and nations on six continents, as well as the thousands of players and fans that these groups represent.
                </Text>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
    <View
      onLayout={[Function]}
      pointerEvents="box-none"
    >
      <View
        collapsable={undefined}
        style={
          Object {
            "backgroundColor": "#F7F7F7",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <View
          collapsable={undefined}
          onLayout={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "backgroundColor": "#F7F7F7",
              "borderBottomColor": "#A7A7AA",
              "borderBottomWidth": 0.5,
              "height": 64,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 20,
            }
          }
        >
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "bottom": 0,
                  "flexDirection": "row",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                collapsable={undefined}
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "left": 70,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 70,
                    "top": 0,
                  }
                }
              >
                <Text
                  accessibilityTraits="header"
                  accessible={true}
                  allowFontScaling={true}
                  collapsable={undefined}
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    Object {
                      "color": "rgba(0, 0, 0, .9)",
                      "fontSize": 17,
                      "fontWeight": "700",
                      "marginHorizontal": 16,
                      "textAlign": "center",
                    }
                  }
                >
                  About Screen
                </Text>
              </View>
              <View
                collapsable={undefined}
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "flexDirection": "row",
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <View
                  accessibilityComponentType="button"
                  accessibilityLabel={undefined}
                  accessibilityTraits={
                    Array [
                      "button",
                    ]
                  }
                  accessible={true}
                  collapsable={undefined}
                  hasTVPreferredFocus={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <View
                    style={
                      Array [
                        Object {},
                      ]
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      disabled={undefined}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {
                            "color": "#007AFF",
                            "fontSize": 18,
                            "padding": 8,
                            "textAlign": "center",
                          },
                        ]
                      }
                    >
                      Back To Home
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
